backend/penndata/views.py

```diff
@@ -416,7 +416,41 @@ def get(self, request, room_id):
                 "usage": {i: amt for i, amt in enumerate(usage_per_hour)},
             }
         )
+    
+class FitnessPreferences(APIView):
+    """
+    GET: returns list of a User's fitness preferences
+
+    POST: updates User fitness preferences by clearing past preferences
+    and resetting them with request data
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+
+        preferences = request.user.profile.fitness_preferences.all()
+
+        # returns all ids in a person's preferences
+        return Response({"rooms": preferences.values_list("id", flat=True)})
+
+    def post(self, request):
+
+        profile = request.user.profile
+
+        # clears all previous preferences in many-to-many
+        profile.fitness_preferences.clear()
+
+        ids = request.data["rooms"]
```

Comment:
make sure to check that the parameter is passed in before using it. a simple way to do this would be to do `request.data.get('rooms').... if ids is None: return <error stuff>`, but you can also first check `if rooms not in request.data: do <error stuff>`